core/src/main/scala/com.softwaremill.kmq/redelivery/Redeliverer.scala

```diff
@@ -55,8 +57,15 @@ class DefaultRedeliverer(
         throw new IllegalStateException(s"Cannot redeliver $marker from topic ${config.getMsgTopic}, due to data fetch timeout")
 
       case Some(toSend) =>
-        logger.info(s"Redelivering message from ${config.getMsgTopic}, partition ${marker.getPartition}, offset ${marker.getMessageOffset}")
-        producer.send(new ProducerRecord(toSend.topic, toSend.partition, toSend.key, toSend.value))
+        val redeliveryCount: Byte = toSend.headers.asScala.find(_.key() == config.getRedeliveryCountHeader).map(_.value()).map(decodeByte).getOrElse(0)
```

Comment:
I wonder if there is any saving with Byte... if we go with Byte, I would adjust the config accordingly.

Reply:
ok, changing to Int